regularizers.py

```diff
@@ -0,0 +1,21 @@
+from keras import regularizers as kr
+from keras import backend as K
+import numpy as np
+import pdb
+
+class L1L2_with_intercept(kr.L1L2):
+    def __init__(self,**kwargs):
+        super().__init__(**kwargs)
+
+    def __call__(self,x):
+        regularization = 0.
+        if self.l1:
+            regularization += K.sum(self.l1 * K.abs(x[:-1]))
+        if self.l2:
+            regularization += K.sum(self.l2 * K.square(x[:-1]))
+        return regularization
+
+def l1_with_inter(l=0.01):
+    # L1 regularizer without regularizing the last column
+    # the last column of the feature represents intercept in this case
+    return L1L2_with_intercept(l1=l)
```

Comment:
You could also use a Dense layer with bias=True, in which case the Dense layer's .kernel will be the weights and its .bias will be the intercept.

Reply:
Right. But with weighted linear regression, the intercept should also be weighted. If bias term is added after I weight the data, it becomes the intercept of a different model. For example, instead of y = beta X + intercept, I would be getting wy = beta wX + intercept_1 which is not what I want.

Reply:
Got it.  You might also look into keras's sample weighting in the fit() function.  I'm not sure how that would interact with the L1 shrinkage.